src/thunks.js

```diff
@@ -42,32 +42,41 @@ export function makeWSConnection() {
     let intervalId;
     let wsConnection = new WebSocket(url);
 
-    wsConnection.onopen = (e) => {
-      wsConnection.send(JSON.stringify({
-        command: 'authenticate',
-        token: authToken
-      }));
-    };
-
     wsConnection.onerror = (e) => {
       dispatch(wsConnectionFailed());
     };
 
+    /*
+    Once a connection is opened, we wait for a "Ready" frame from the Server.
+    This is to avoide sending an Auth message while the server still have not 
+    had its on-receive callback hooked up.
+    Reference:- ttps://github.com/http-kit/http-kit/issues/318 
+    */
     wsConnection.onmessage = (e) => {
       let message = JSON.parse(e.data);
-      if (message['auth-status'] === 'success') {
-        dispatch(wsReceivedMessage(JSON.parse(e.data)));
-        dispatch(wsConnectionReady(wsConnection));
-        // Begin ping-pong at 10s intervals.
-        intervalId = window.setInterval(() => sendPing(wsConnection), 10000);
-        // Redefine onmessage to stop checking authentication status.
-        wsConnection.onmessage = (e) => {
-          dispatch(wsReceivedMessage(JSON.parse(e.data)));
-        }
+
+      if (message.type === 'ready') {
+          wsConnection.send(JSON.stringify({
+          command: 'authenticate',
+          token: authToken
+        })); 
       }
       else {
-        dispatch(wsHandshakeFailed());
+        if (message['auth-status'] === 'success') {
```

Comment:
@trycatcher here if the first message is a `success` message and not a ready frame, the connection will still be considered authenticated and ready, even though the client never sent an authentication message.

Although this 'works', I think our implementation ought to be more correct. The client should only wait for the authentication success message after it has received the `ready` message and sent the `authenticate` message. Here's a state diagram to clarify:

waiting for ready frame --> waiting for authentication success --> receiving data messages